### PR TITLE
Leave/Rejoin notification

### LINKED ISSUE #50: Add notification on player leave / reconnect on versus mode
Currently, there is no way for someone playing the game to know whether or not the other player left/reconnected to the instance. This can be pretty simply implemented into the chat as a system message (for which there is already an existing implementation).

### PR DESCRIPTION
**Pull request by arjunghoshal**
Resolved #50

### CODE REVIEW

File: source/server/main.js [hunk 1]
```diff
@@ -453,7 +461,13 @@ function handleInitialization(webSocketConnection, playerUUID) {
   }
 
   const attemptRejoinStatus = attemptRejoin(webSocketConnection, playerUUID);
-  if (!attemptRejoinStatus) createInstance(webSocketConnection);
+  if (!attemptRejoinStatus) {
```
Comment (Luke-Sheltraw) [inline source/server/main.js:464]: an if/else statement should probably have the negation as the else clause instead of the if clause, for clarity's sake

**Review by Luke-Sheltraw — CHANGES_REQUESTED** [review]

File: source/server/main.js [hunk 2]
```diff
@@ -380,18 +381,25 @@ function handleStartRound(webSocketConnection) {
...
  * @param { string } messageContents text content of message being received
+ * @param { boolean } systemMessage whether the message should be sent as a system message
  */
-function handleChatMessage(webSocketConnection, messageContents) {
+function handleChatMessage(webSocketConnection, messageContents, systemMessage = false) {
   const gameInstance =
     gameInstancesByPlayerUUID[webSocketConnection.profile.uuid];
 
   // TODO: validate chat message contents
+  let profile = structuredClone(webSocketConnection.profile);
```
[the remark this answers is not in the document]
Comment (arjunghoshal) [reply source/server/main.js:391]: Sounds good to me, I'll implement that with the disconnect/leave stuff. Should we update the actions flowchart to add the new action?

[the remark this answers is not in the document]
Comment (Luke-Sheltraw) [reply source/server/main.js:391]: There are actually a few actions that need to be added to the flow chart lol

**Review by Luke-Sheltraw — APPROVED** [review]
Thanks for making the changes

**Pull request merged**